src/node.cc

```diff
@@ -4473,6 +4492,11 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
   if (debug_enabled)
     EnableDebug(&env);
 
+  if (load_napi_modules) {
+    ProcessEmitWarning(&env, "N-API is an experimental feature "
+        "and could change at any time.");
+  }
+
```

Comment:
VM Summit 3 [meeting notes](https://github.com/joshgav/abi-stable-node/blob/f7fc7e4bddd222aba922d52f4447bcab2fd40691/meeting%20notes/2017-03-03.md) has this 
_For 8.0, include an opt-in runtime flag, show experimental warning when a NAPI module is loaded._